admin_scripts/aula_register.py

```diff
@@ -0,0 +1,205 @@
+#%    If it is not provided the aula default register url is used.
+#%    It takes one mandatory parameter: the register code needed to activate the "kommegaa" screen.
+#%
+#================================================================
+#- IMPLEMENTATION
+#-    version         aula_register.py (magenta.dk) 1.0.0
+#-    author          Danni Als
+#-    copyright       Copyright 2019, Magenta ApS
+#-    license         GNU General Public License v3+
+#-    email           danni@magenta.dk
+#-
+#================================================================
+#  HISTORY
+#     2019/10/28 : af : Script created
+#
+#================================================================
+# END_OF_HEADER
+#================================================================
+
+import os
+import glob
+import sys
+import stat
+import datetime
+import subprocess
+
+subprocess.call([sys.executable, "-m", "pip", "install", 'wget'])
+subprocess.call([sys.executable, "-m", "pip", "install", 'selenium'])
+
+print('Installed wget and selenium.')
+
+import zipfile
+import sqlite3
+
+from selenium.webdriver import Chrome
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as expected
+from selenium.common.exceptions import InvalidArgumentException
+
+if len(sys.argv) == 3:
+    url = sys.argv[1]
+    print('URL: {}'.format(url))
+    register_code = sys.argv[2]
+    print('Registreringsskode: {}'.format(register_code))
+    try:
+        int(register_code)
+    except ValueError:
+        print('Registeringskode er ikke et tal.')
+        sys.exit(1)
+    if len(register_code) != 8:
+        print('Registeringskode har ikke den korrekte længde.')
+        sys.exit(1)
+else:
+    print('Der mangler en eller flere input parametre.')
+    sys.exit(1)
+
+# get chrome version
+chrome_version = os.popen('google-chrome --version').read()
+# String returned is always 'Google Chrome xx.x.xxxx.xx'
+chrome_version = chrome_version.split(' ')
+chrome_version = chrome_version[2].split('.')[0]
+print('Chrome version installed: {}'.format(chrome_version))
+if chrome_version == '74':
+    driver_version = '74.0.3729.6'
+elif chrome_version == '75':
+    driver_version = '75.0.3770.140'
+elif chrome_version =='76':
+    driver_version = '76.0.3809.126'
+elif chrome_version == '77':
+    driver_version = '77.0.3865.40'
+elif chrome_version == '78':
+    driver_version = '78.0.3904.70'
+else:
+    print('Chrome version not supported.')
+    sys.exit(1)
+
+system_path = '/usr/local/bin'
+zipfile_name = driver_version + 'chromedriver_linux64.zip'
+zip_path = os.path.join(system_path, zipfile_name)
+extracted_filename = 'chromedriver'
+extracted_filepath = os.path.join(system_path, extracted_filename)
+
+# download gecko and setup
```

Comment:
(A minor and trivial observation, but we're not downloading _Gecko_ for Chromium-based browsers.)